### ISSUE TITLE
Geometry binary data appears to be incorrect value

### ISSUE THREAD

**Issue by dylel** [planetscale/database-js#98]
For some context i'm trying to interface with drizzle orm. I'm trying to parse the response for a geometry field that I'm receiving from the driver, i'm deliberately not using ST_AsBinary() as the will be how it will end up getting called by drizzle. My understanding is that MySql should return the SRID + WKB as the default response when not providing ST_AsText, or ST_AsBinary

The geometry in my database: LINESTRING(0 0,1 1,2 2), 4326

```
db.Execute('SELECT Top(1) geom from polys')
```

The response from the driver:
```
\u0010\u0000\u0000\u0001\u0002\u0000\u0000\u0000\u0003\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000\u0000 ?\u0000\u0000\u0000\u0000\u0000\u0000 ?\u0000\u0000\u0000\u0000\u0000\u0000\u0000@\u0000\u0000\u0000\u0000\u0000\u0000\u0000@
```

I'm then reading it into a buffer and slicing off the first 4 bytes for the SRID. However doing so produces a marginally different output than expected. 

I'm expecting (from MySqlWorkBench):

```
01020000000300000000000000000000000000000000000000000000000000F03F000000000000F03F00000000000000400000000000000040
```

Receiving (after reading from buffer):

```
01020000000300000000000000000000000000000000000000000000000000fd3f000000000000fd3f00000000000000400000000000000040
```
its very close to what it should be

MySqlWorkBench for reference:
![image](https://user-images.githubusercontent.com/44506538/229333539-474a8015-6489-4c52-86b9-ac8a3e22584e.png)

Is there something i'm missing here?

[Codesandbox without pscale hooked up](https://codesandbox.io/p/sandbox/vibrant-orla-jhh1hm?file=%2Fsrc%2Findex.ts&selection=%5B%7B%22endColumn%22%3A1%2C%22endLineNumber%22%3A26%2C%22startColumn%22%3A1%2C%22startLineNumber%22%3A26%7D%5D)

**Comment by mattrobenolt** [comment]
That's interesting. I wonder if this is a bug in Vitess. I'll help try and investigate.

**Comment by dylel** [comment]
Awesome, thanks for that. I was looking into how mysql deals with and testing that to make sure everything checks out. However i've fired this up this morning and i'm now getting the correct values, did you work your magic on this?

**Comment by dylel** [comment]
Think i've figured out where it's coming from, i've added my own blank type cast for a geometry field:

```
 // create the connection
  const connection = connect({
    url: env.DATABASE_URL,
    cast: (field, value) => {
      if (field.type === "geometry") {
      }
      return value;
    },
  });
```

and now i receive the exact same hex as expected when reading from the buffer and can successfully extract the srid and it all matches what its supposed to be. So i suspect that the driver must be slightly altering the geometry field inside the default cast function

**Comment by mattrobenolt** [comment]
Oh, you're right. Our default case is to decode the binary into text. Which is likely why this is happening. We should handle this and explicitly return the raw binary.

Thanks for pointing this out! I definitely don't have much experience with geometry types personally.

**Comment by nmajor** [comment]
Following closely in the footsteps of @dylel, I was able to get geometry columns working with drizzle-orm and database-js with this code for my casting:

```
const connection = connect({
  url: env.DATABASE_URL,
  cast: (field, value) => {
    if (field.type === "GEOMETRY") {
      return value;
    }
    return cast(field, value);
  },
});
```

Later in the flow, I also had to manually remove the first 4 bytes from the value before using the wkx lib to convert it into geoJson.

Thanks @dylel

**Comment by iheanyi** [comment]
Hey y'all, we've fixed a bug with this in [v1.8.0](https://github.com/planetscale/database-js/releases/tag/v1.8.0). This now returns the geometry data as a raw value, letting the caller handle it!
